services/QuillLMS/client/app/bundles/Grammar/actions/session.ts

```diff
@@ -132,7 +132,16 @@ const activityUID = (): string => { return getParameterByName('uid', window.loca
 const handleGrammarSession = (session) => {
   return dispatch => {
     if (session && Object.keys(session).length > 1 && !session.error) {
-      QuestionApi.getAllForActivity(activityUID()).then((allQuestions) => {
+
+      // activityUID will not be set in cases where a user is resuming a
```

Comment:
Question: did we confirm that not having questions populated was causing our error? i.e. could we return an empty object, or maybe just return one question? It seems that proofreader activities do not use questions at all, so it seems wasteful to pull all the questions here.

Reply:
What's happening is that if you don't get all the questions right on the Proofreader activity, the system attempts to start the follow-up Grammar activity.  But the uid is still not set in the URL, so the Grammar activity is what's crashing.